angular-view/index.js

```diff
@@ -0,0 +1,138 @@
+    }];
+
+    // Add module choices
+    if (fs.existsSync(modulesFolder)) {
+
+      fs.readdirSync(modulesFolder).forEach(function (folder) {
+        var stat = fs.statSync(modulesFolder + '/' + folder);
+
+        if (stat.isDirectory()) {
+          prompts[0].choices.push({
+            value: folder,
+            name: folder
+          });
+        }
+      });
+    }
+
+    this.prompt(prompts, function (props) {
+      this.moduleName = props.moduleName;
+      this.name = props.name || this.moduleName;
+
+      this.controllerName = props.controllerName;
+
+      this.slugifiedModuleName = s(this.moduleName).slugify().value();
+      this.humanizedModuleName = s(this.moduleName).humanize().value();
+
+      this.slugifiedName = s(this.name).humanize().slugify().value();
+      this.classifiedName = s(this.slugifiedName).classify().value();
+      this.humanizedName = s(this.slugifiedName).humanize().value();
+
+      done();
+    }.bind(this));
+  },
+
+  askForControllerName: function () {
+    var done = this.async();
+
+    var prompts = [{
+      name: 'controllerName',
+      message: 'What is the name of the controller this view will use?',
+      default: this.classifiedName
+    }];
+
+    this.prompt(prompts, function (props) {
+      this.controllerName = props.controllerName;
+
+      this.slugifiedControllerName = s(this.controllerName).slugify().value();
+      this.classifiedControllerName = s(this.slugifiedControllerName).classify().value();
+
+      done();
+    }.bind(this));
+  },
+
+  askToAddRoute: function () {
+    var done = this.async();
+
+    var prompts = [{
+      type: 'confirm',
+      name: 'addRoute',
+      message: 'Would you like to add a route for this view?',
+      default: true
+    }];
+
+    this.prompt(prompts, function (props) {
+      this.addRoute = props.addRoute;
+
+      done();
+    }.bind(this));
+  },
+
+  askForRouteDetails: function () {
+    if (this.addRoute) {
+      var done = this.async();
+
+      var prompts = [{
+        name: 'routePath',
+        message: 'What is your view route path?',
+        default: this.slugifiedName
+      }];
+
+      this.prompt(prompts, function (props) {
+        this.routePath = props.routePath;
+        this.slugifiedRoutePath = s(this.routePath).slugify().value();
+
+        done();
+      }.bind(this));
+    }
+  },
+
+  renderRoute: function () {
+    if (this.addRoute) {
+      var routesFilePath = process.cwd() + '/modules/' + this.slugifiedModuleName + '/client/config/' + this.slugifiedModuleName + '.client.routes.js';
+
+      // If routes file exists we add a new state otherwise we render a new one
+      if (fs.existsSync(routesFilePath)) {
+        // Read the source routes file content
+        var routesFileContent = this.readFileAsString(routesFilePath);
+
+        // Append the new state
+        routesFileContent = routesFileContent.replace('$stateProvider.', engine(this.read('_.client.route.js'), this));
```

Comment:
Before, 'this.engine' was used (that very same 'ejs') for rendering
template into a string. Now, in later version of 'yo', 'this.engine' was
removed, so I added it as dependency.

On Friday, 8 January 2016, Ilan Biala <notifications@github.com> wrote:

> In angular-view/index.js
> <https://github.com/meanjs/generator-meanjs/pull/165#discussion_r49158473>
> :
>
> > +        done();
> > +      }.bind(this));
> > +    }
> > +  },
> > +
> > +  renderRoute: function () {
> > +    if (this.addRoute) {
> > +      var routesFilePath = process.cwd() + '/modules/' + this.slugifiedModuleName + '/client/config/' + this.slugifiedModuleName + '.client.routes.js';
> > +
> > +      // If routes file exists we add a new state otherwise we render a new one
> > +      if (fs.existsSync(routesFilePath)) {
> > +        // Read the source routes file content
> > +        var routesFileContent = this.readFileAsString(routesFilePath);
> > +
> > +        // Append the new state
> > +        routesFileContent = routesFileContent.replace('$stateProvider.', engine(this.read('_.client.route.js'), this));
>
> Why do you need ejs?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/meanjs/generator-meanjs/pull/165/files#r49158473>.
>